### PR TITLE
Add content for first draft

### PR DESCRIPTION
**Pull request by thomastuts**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -6,22 +6,43 @@
...
-course), and side effects in when rendering can grind an application to a halt.
-
-At the same time, JavaScript is fast while the DOM APIs are slow — this point is
-worth emphasizing.
+## What's a virtual DOM?
+![React VR?](https://media.giphy.com/media/3o7qDL7l1IZAQvxvj2/giphy.gif)
+
+By now, you should already know what the DOM is: a programmatic representation of the document we see in the browser.
+So, what's all this virtual DOM ruckus about? Well, it turns out that our good 'ol friend DOM is slow — terribly slow. He
+makes a granny using a walker look like Usain Bolt.
```
Comment (ezmiller) [inline README.md]: Just a gentle, sincere question: might it be wise to avoid the granny reference? Another comparison?

File: README.md [hunk 2]
```diff
@@ -6,22 +6,43 @@
...
+Virtual DOM is a technique employed by several front-end libraries and frameworks, most notably React. In a nutshell,
+virtual DOM builds a _virtual_ representation of what our document should look like. When we're ready to render things
+to the screen, the virtual DOM will take a look at the existing DOM and change only what needs to be changed (in more
+technical terms, it's diffing and re-rendering the changes).
+
+The performance gains here are not to be underestimated, due to the way the virtual DOM works. Virtual DOM is created by
+a bunch of super smart people using extremely efficient algorithms. Most virtual DOMs also batch their updates to the
+DOM, ensuring that the _real_ DOM gets modified as little as possible. The point is: it's best to touch and change the
+DOM as little as possible, and when we do, we do it with surgical precision. Think of virtual DOM as our personal DOM
+surgeon!
```
Comment (ezmiller) [inline README.md]: Very nice explanation! 👍

File: README.md [hunk 3]
```diff
@@ -6,22 +6,43 @@
...
+
+## So why use it?
+The above section should have already given you a good idea of when a virtual DOM is useful. It's a particularly great
+choice when we're building apps where changes in the interface happen a _lot_. By reducing the load on the real DOM, we
```
Comment (ezmiller) [inline README.md:28]: "a lot" -> "frequently" ? (less colloquial?)

File: README.md [hunk 4]
```diff
@@ -6,22 +6,43 @@
...
+can make our apps behave much more fluently and in line with what a user expects from software nowadays.
+
+In virtual DOM, we can just say 'okay, our document should look like _this_' — and the virtual DOM will do the heavy
+lifting for us (i.e. diffing the changes and applying them). This means that we don't need to write imperative code to
+update every tiny bit of our application, we just declare what the end result should look like, and the virtual DOM will
+do the rest!
```
Comment (ezmiller) [inline README.md]: Really important point! I think you could, if you wanted, even emphasize it more: "The ability to simply **declare** what the DOM should look and how it should behave is a huge win. It means that we can think about our front-end interfaces at a much higher level of abstraction that pure html." You could even foreshadow the component idea...  But it's totally fine as is as well.

File: README.md [hunk 5]
```diff
@@ -6,22 +6,43 @@
...
+
+These abstractions also paved the road for non-DOM implementations for virtual DOM, for example React Native. In this
+case, it's easier to think of DOM as a generic document, rather than the thing we see in our web browser.
+
+## What's the catch?
+Even though virtual DOM is very performant and used in several high-profile libraries and frameworks, it's important to
+remember that it's still a _really_ clever workaround. There are other solutions in the works (like Shadow DOM) by
+browser vendors to make the DOM faster, but for right now, virtual DOM seems to be our best bet.
```
Comment (ezmiller) [inline README.md]: Just a minor question, what's the difference between referring to "the Virtual DOM" and just "virtual DOM". Is one more correct than the other?

File: README.md [hunk 6]
```diff
@@ -6,22 +6,43 @@
...
 
 ## Resources
-
 - [Why did we build React?](https://facebook.github.io/react/blog/2013/06/05/why-react.html)
+- [The difference between Virtual DOM and DOM](http://reactkungfu.com/2015/10/the-difference-between-virtual-dom-and-dom/)
 - [React (Virtual) DOM Terminology](https://facebook.github.io/react/docs/glossary.html)
```
Comment (ezmiller) [inline README.md:48]: Nice resources!

File: README.md [hunk 7]
```diff
@@ -6,22 +6,43 @@
...
+lifting for us (i.e. diffing the changes and applying them). This means that we don't need to write imperative code to
+update every tiny bit of our application, we just declare what the end result should look like, and the virtual DOM will
+do the rest!
+
+These abstractions also paved the road for non-DOM implementations for virtual DOM, for example React Native. In this
+case, it's easier to think of DOM as a generic document, rather than the thing we see in our web browser.
+
+## What's the catch?
+Even though virtual DOM is very performant and used in several high-profile libraries and frameworks, it's important to
+remember that it's still a _really_ clever workaround. There are other solutions in the works (like Shadow DOM) by
```
Comment (ezmiller) [inline README.md]: "it's still" -> "it's still **just** a". Also It's a clever workaround to which problem? The slow DOM? If so, maybe just a quick reminder.

File: README.md [hunk 8]
```diff
@@ -6,22 +6,43 @@
...
+In virtual DOM, we can just say 'okay, our document should look like _this_' — and the virtual DOM will do the heavy
+lifting for us (i.e. diffing the changes and applying them). This means that we don't need to write imperative code to
+update every tiny bit of our application, we just declare what the end result should look like, and the virtual DOM will
+do the rest!
+
+These abstractions also paved the road for non-DOM implementations for virtual DOM, for example React Native. In this
+case, it's easier to think of DOM as a generic document, rather than the thing we see in our web browser.
+
+## What's the catch?
+Even though virtual DOM is very performant and used in several high-profile libraries and frameworks, it's important to
```
Comment (ezmiller) [inline README.md]: Technically, "performant" is not a word, though it's seen a lot of actual usage obviously related to the rise of software. [See here](https://books.google.com/ngrams/graph?content=performant&year_start=1800&year_end=2000&corpus=15&smoothing=3&share=&direct_url=t1%3B%2Cperformant%3B%2Cc0). But it really is not a word, so might be wise to use just "perform" which is also a pretty good word. ;)

File: README.md [hunk 9]
```diff
@@ -6,22 +6,43 @@
...
+By now, you should already know what the DOM is: a programmatic representation of the document we see in the browser.
+So, what's all this virtual DOM ruckus about? Well, it turns out that our good 'ol friend DOM is slow — terribly slow. He
+makes a granny using a walker look like Usain Bolt.
+
+Virtual DOM is a technique employed by several front-end libraries and frameworks, most notably React. In a nutshell,
+virtual DOM builds a _virtual_ representation of what our document should look like. When we're ready to render things
+to the screen, the virtual DOM will take a look at the existing DOM and change only what needs to be changed (in more
+technical terms, it's diffing and re-rendering the changes).
+
+The performance gains here are not to be underestimated, due to the way the virtual DOM works. Virtual DOM is created by
```
Comment (ezmiller) [inline README.md]: This sentence confuses me a bit: "The performance gains here are not to be underestimated, due to the way the Virtual DOM works." Seems like you are saying that due to the implementation we shouldn't underestimate. I think just: "The performance gains here are not to be underestimated" will do well! And you are right!

File: README.md [hunk 10]
```diff
@@ -6,22 +6,43 @@
 2. Describe why a virtual DOM might be useful
 3. Describe some of the tradeoffs of using a virtual DOM
```
Comment (ezmiller) [inline README.md:7]: Do you achieve this objective in the document? Is the fact that it is a workaround, as you say at the end, the tradeoff?

**Comment by ezmiller** [comment]
Okay @thomastuts once you consider these comments, I think this is ready to go. It's very well done. Most of my comments were just about polishing.

File: README.md (hunk 5 above)
Comment (thomastuts) [reply README.md]: Honestly, I have no idea — both seem to be used interchangeably in other resources too (sometimes in the same blogpost/guide). Guess we should just pick one and be consistent 😄

**Pull request merged**